src/swaps/blindex/BlindexSwapProvider.js

```diff
@@ -0,0 +1,509 @@
+    this.availableTokenSymbols = blindexConfig.ERC20_INFO.filter((entry) => entry !== 'UNI-V2').map(
+      (entry) => (entry.symbol === 'WRBTC' ? this.nativeTokenSymbol : entry.symbol)
+    )
+  }
+
+  async getSupportedPairs() {
+    return []
+  }
+
+  getApi(network, asset) {
+    const fromChain = cryptoassets[asset].chain
+    const chainId = ChainNetworks[fromChain][network].chainId
+    if (chainId in this._apiCache) {
+      return this._apiCache[chainId]
+    } else {
+      const api = new ethers.providers.StaticJsonRpcProvider(RSK_RPC_URL)
+      this._apiCache[chainId] = api
+      return api
+    }
+  }
+
+  getToken(chainId, asset) {
+    const assetData = cryptoassets[asset]
+    if (asset === this.nativeTokenSymbol) {
+      assetData.contractAddress = this.wrbtcAddress
+    }
+
+    return new Token(
+      chainId,
+      assetData.contractAddress,
+      assetData.decimals,
+      assetData.code,
+      assetData.name
+    )
+  }
+
+  getMinimumOutput(outputAmount) {
+    const slippageTolerance = new Percent('200', '10000') // 2%
+    const slippageAdjustedAmountOut = new Fraction(JSBI.BigInt(1))
+      .add(slippageTolerance)
+      .invert()
+      .multiply(outputAmount.quotient).quotient
+    return CurrencyAmount.fromRawAmount(outputAmount.currency, slippageAdjustedAmountOut)
+  }
+
+  /**
+   * Get a quote for the specified parameters
+   * @param {{ network, from, to, amount }} options
+   */
+  // eslint-disable-next-line no-unused-vars
+  async getQuote({ network, from, to, amount }) {
+    if (
+      network != 'mainnet' ||
+      !this.availableTokenSymbols.includes(from) ||
+      !this.availableTokenSymbols.includes(to)
+    ) {
+      return null
+    }
+
+    const fromContractAddress =
+      from === this.nativeTokenSymbol ? this.wrbtcAddress : cryptoassets[from].contractAddress
+    const toContractAddress =
+      to === this.nativeTokenSymbol ? this.wrbtcAddress : cryptoassets[to].contractAddress
+    const bestRouteWithValOut = await this.calcBestRoute(
+      network,
+      to,
+      amount,
+      false,
+      fromContractAddress,
+      toContractAddress
+    )
+    const amountOut_d18 = this.ethersBigNumberToDecimal(bestRouteWithValOut.finalAmount, DECIMALS)
+    return {
+      from,
+      to,
+      fromAmount: currencyToUnit(cryptoassets[from], BN(amount)),
+      toAmount: currencyToUnit(cryptoassets[to], BN(amountOut_d18))
+    }
+  }
+
+  /**
+   * Create a new swap for the given quote
+   * @param {{ network, walletId, quote }} options
+   */
+  // eslint-disable-next-line no-unused-vars
+  async newSwap({ network, walletId, quote }) {
+    const approvalRequired = isERC20(quote.from)
+    const updates = approvalRequired
+      ? await this.approveTokens({ network, walletId, quote })
+      : await this.sendSwap({ network, walletId, quote })
+
+    return {
+      id: uuidv4(),
+      fee: quote.fee,
+      slippage: 200,
```

Comment:
This is the second time we're defining the slippage. Can we have it in one place and reuse it?
So if we'll change it in the future we won't forget to update all the places

Reply:
done